### PR TITLE
Suggested changes for ansible:

### PR DESCRIPTION
**Pull request by josejulio**
- Remove code from aggregator
 - Use ansible as the bundle and reports as the app +
 - Add instant body and title

### CODE REVIEW

File: backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/AnsibleEmailAggregator.java [hunk 1]
```diff
@@ -10,117 +10,14 @@
...
-                }
-                if (currentDisruptionImpactAvailability || hasAvailabilityImpact) {
-                    impacts.add(AVAILABILITY);
-                }
-
-                String newImpacts = impacts.stream().collect(Collectors.joining(", "));
-                serviceDisruption.put(SERVICE_DISRUPTION_IMPACT, newImpacts);
-            }
-        });
+        throw new RuntimeException("Not implemented yet");
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/AnsibleEmailAggregator.java:20]: I'm removing the code from this file. You are not going to need it for now.

File: backend/src/main/java/com/redhat/cloud/notifications/templates/Ansible.java [hunk 2]
```diff
@@ -10,10 +10,8 @@
     @Override
     public TemplateInstance getTitle(String eventType, EmailSubscriptionType type) {
         if (type == EmailSubscriptionType.INSTANT) {
-            if (eventType.equals("new-recommendation")) {
-                return Templates.newRecommendationInstantEmailTitle();
-            } else if (eventType.equals("resolved-recommendation")) {
-                return Templates.resolvedRecommendationInstantEmailTitle();
+            if (eventType.equals("report-available")) {
+                return Templates.instantEmailTitle();
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/templates/Ansible.java:14]: Linking your `report-available` event to a title

File: backend/src/main/java/com/redhat/cloud/notifications/templates/Ansible.java [hunk 3]
```diff
@@ -26,10 +24,8 @@ public TemplateInstance getTitle(String eventType, EmailSubscriptionType type) {
     @Override
     public TemplateInstance getBody(String eventType, EmailSubscriptionType type) {
         if (type == EmailSubscriptionType.INSTANT) {
-            if (eventType.equals("new-recommendation")) {
-                return Templates.newRecommendationInstantEmailBody();
-            } else if (eventType.equals("resolved-recommendation")) {
-                return Templates.resolvedRecommendationInstantEmailBody();
+            if (eventType.equals("report-available")) {
+                return Templates.instantEmailBody();
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/templates/Ansible.java:28]: Linking your `report-available` event to an html body

File: backend/src/main/java/com/redhat/cloud/notifications/templates/Ansible.java [hunk 4]
```diff
@@ -41,7 +37,7 @@ public TemplateInstance getBody(String eventType, EmailSubscriptionType type) {
 
     @Override
     public boolean isSupported(String eventType, EmailSubscriptionType type) {
-        return (eventType.equals("new-recommendation") || eventType.equals("resolved-recommendation")) && type == EmailSubscriptionType.INSTANT;
+        return eventType.equals("report-available") && type == EmailSubscriptionType.INSTANT;
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/templates/Ansible.java:40]: Only the `instant` event for `report-available` is supported.

File: backend/src/main/java/com/redhat/cloud/notifications/templates/EmailTemplateFactory.java [hunk 5]
```diff
@@ -39,6 +39,10 @@ public EmailTemplate get(String bundle, String application) {
             if (application.equalsIgnoreCase(RHOSAK)) {
                 return new Rhosak();
             }
+        } else if (bundle.equalsIgnoreCase(BUNDLE_ANSIBLE)) {
+            if (application.equalsIgnoreCase(APP_ANSIBLE_REPORTS)) {
+                return new Ansible();
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/templates/EmailTemplateFactory.java:44]: Moved to bundle: ansible / app: reports

File: backend/src/main/resources/templates/Ansible/instantEmailBody.html [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{#include Ansible/insightsEmailBody}
```
Comment (josejulio) [inline backend/src/main/resources/templates/Ansible/instantEmailBody.html:1]: Added a basic email template you can start using

File: backend/src/main/resources/templates/Ansible/instantEmailBody.html [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+{#content-body}
+<tr>
+    <td class="rh-content__block">
+        <p>Ansible report available for download at {action.context.system_check_in.toUtcFormat()}</p>
+    </td>
+</tr>
+<tr>
+    <td class="rh-content__block">
+        <div class="rh-stack">
+            <a class="rh-url" href="https://console.redhat.com/insights/inventory/{action.context.report_id}"><b>Download now</b></a>
```
Comment (josejulio) [inline backend/src/main/resources/templates/Ansible/instantEmailBody.html]: You will need to update the URL here.
I'm considering that you will send the report_id or something in the context. Let me know what you expect to send and I can guide you on what we need

Comment (h-kataria) [reply backend/src/main/resources/templates/Ansible/instantEmailBody.html]: this link should be `https://console.redhat.com/ansible/insights/reports/{action.context.slug}`

Comment (josejulio) [reply backend/src/main/resources/templates/Ansible/instantEmailBody.html]:
```suggestion
            <a class="rh-url" href="https://console.redhat.com/ansible/insights/reports/{action.context.slug}"><b>Download now</b></a>
```

**Review by h-kataria — APPROVED** [review]

**Pull request merged**